plugins/modules/gcp_compute_subnetwork.py

```diff
@@ -353,7 +354,10 @@ def main():
 
     if fetch:
         if state == 'present':
-            if is_different(module, fetch):
+            if module.params['network']['selfLink'] != fetch['network']: # found difference on same subnet within the same VPC network
+                module.fail_json(msg="Subnet already exists in a different VPC network: %s, please change the name or region" % fetch['network'])
```

Comment:
```suggestion
                module.fail_json(msg="Subnet already exists in a different VPC network: %s" % fetch['network'])
```

I feel like the guidance on changing the name or region might be case-specific, so perhaps best just to state the error and let the user decide how to resolve it?